problem/src/main/java/org/zalando/problem/Status.java

```diff
@@ -268,6 +270,19 @@
         this.reason = reasonPhrase;
     }
 
+    /**
+     * Creates a Status instance from the given code.
+     *
+     * @param code the HTTP code as a number
+     * @return the correct enum value for this status code.
+     */
+    public static Status ofCode(int code) {
+        return Arrays.stream(Status.values())
```

Comment:
Personally I'm not a big fan of linear search. What about constructing a map in a static initializer which we just use here?

Reply:
I'm not used to do that but I can see why it'd be better in most cases. I'll make the change.